core/chains/evm/logpoller/log_poller.go

```diff
@@ -172,30 +157,30 @@ func (lp *logPoller) Close() error {
 func (lp *logPoller) run() {
 	defer close(lp.done)
 	tick := time.After(0)
-	var start int64
+	var replay int64
 	for {
 		select {
 		case <-lp.ctx.Done():
 			return
 		case fromBlock := <-lp.replay:
 			lp.lggr.Warnw("Replay requested", "from", fromBlock)
-			start = fromBlock
+			replay = fromBlock
+			// Immediate replay.
+			tick = time.After(0)
 		case <-tick:
 			tick = time.After(utils.WithJitter(lp.pollPeriod))
-			if start != 0 {
-				start = lp.pollAndSaveLogs(lp.ctx, start)
-				continue
-			}
-			// Otherwise, still need initial start
+			// Always start from the latest block in the db.
+			var start int64
```

Comment:
This definitely simplifies things (which I'm a fan of) however I imagine sometime down the line we'd maybe want to circumvent this, especially if a node is re-started after being down for a while, meaning the latest block that's in the DB is very old. I guess you'd put that logic somewhere here though it'd def "uglify" the logic.

Reply:
re-starting after being down for a while should be ok since we do the backfill in batches. I think it'd be very rare that even that is too slow, but if it really was you could wipe the db and treat it like an initial first connect. Probably can revisit if you have a specific scenario in mind